### ISSUE TITLE
[3.0.1] Relative path is expected, path / is an absolute path.

### ISSUE THREAD

**Issue by boris-petrov** [broccolijs/broccoli-funnel#121]
Doesn't happen on `3.0.0`.

```
ERROR Summary:

  - broccoliBuilderErrorStack: Error: Relative path is expected, path / is an absolute path.
    at Object.handleFs (/home/boris/project/node_modules/broccoli-funnel/node_modules/broccoli-output-wrapper/dist/index.js:27:15)
    at Funnel._copy (/home/boris/project/node_modules/broccoli-funnel/index.js:475:19)
    at Funnel.build (/home/boris/project/node_modules/broccoli-funnel/index.js:235:16)
    at /home/boris/project/node_modules/ember-cli/node_modules/broccoli/dist/wrappers/transform-node.js:68:45
    at new Promise (<anonymous>:null:null)
    at TransformNodeWrapper.build (/home/boris/project/node_modules/ember-cli/node_modules/broccoli/dist/wrappers/transform-node.js:53:16)
    at /home/boris/project/node_modules/ember-cli/node_modules/broccoli/dist/builder.js:99:36

  - code: [undefined]
  - codeFrame: Relative path is expected, path / is an absolute path.
  - errorMessage: Relative path is expected, path / is an absolute path.
        at Funnel
-~- created here: -~-
    at new Plugin (/home/boris/project/node_modules/broccoli-funnel/node_modules/broccoli-plugin/dist/index.js:45:33)
    at new Funnel (/home/boris/project/node_modules/broccoli-funnel/index.js:48:5)
    at new funnel (/home/boris/project/node_modules/broccoli-funnel/index.js:498:10)
    at module.exports (/home/boris/project/ember-cli-build.js:291:5)
    at Builder.readBuildFile (/home/boris/project/node_modules/ember-cli/lib/models/builder.js:52:14)
    at Builder.setupBroccoliBuilder (/home/boris/project/node_modules/ember-cli/lib/models/builder.js:66:22)
    at new Builder (/home/boris/project/node_modules/ember-cli/lib/models/builder.js:32:10)
    at BuildTask.run (/home/boris/project/node_modules/ember-cli/lib/tasks/build.js:15:19)
    at /home/boris/project/node_modules/ember-cli/lib/models/command.js:239:24
    at tryCatcher (/home/boris/project/node_modules/rsvp/dist/rsvp.js:326:21)
    at invokeCallback (/home/boris/project/node_modules/rsvp/dist/rsvp.js:498:33)
    at /home/boris/project/node_modules/rsvp/dist/rsvp.js:562:16
    at flush (/home/boris/project/node_modules/rsvp/dist/rsvp.js:2441:7)
    at processTicksAndRejections (internal/process/task_queues.js:79:11)

-~- (end) -~-
  - errorType: Build Error
  - location:
    - column: [undefined]
    - file: [undefined]
    - line: [undefined]
    - treeDir: [undefined]
  - message: Relative path is expected, path / is an absolute path.
        at Funnel
-~- created here: -~-
    at new Plugin (/home/boris/project/node_modules/broccoli-funnel/node_modules/broccoli-plugin/dist/index.js:45:33)
    at new Funnel (/home/boris/project/node_modules/broccoli-funnel/index.js:48:5)
    at new funnel (/home/boris/project/node_modules/broccoli-funnel/index.js:498:10)
    at module.exports (/home/boris/project/ember-cli-build.js:291:5)
    at Builder.readBuildFile (/home/boris/project/node_modules/ember-cli/lib/models/builder.js:52:14)
    at Builder.setupBroccoliBuilder (/home/boris/project/node_modules/ember-cli/lib/models/builder.js:66:22)
    at new Builder (/home/boris/project/node_modules/ember-cli/lib/models/builder.js:32:10)
    at BuildTask.run (/home/boris/project/node_modules/ember-cli/lib/tasks/build.js:15:19)
    at /home/boris/project/node_modules/ember-cli/lib/models/command.js:239:24
    at tryCatcher (/home/boris/project/node_modules/rsvp/dist/rsvp.js:326:21)
    at invokeCallback (/home/boris/project/node_modules/rsvp/dist/rsvp.js:498:33)
    at /home/boris/project/node_modules/rsvp/dist/rsvp.js:562:16
    at flush (/home/boris/project/node_modules/rsvp/dist/rsvp.js:2441:7)
    at processTicksAndRejections (internal/process/task_queues.js:79:11)

-~- (end) -~-
  - name: BuildError
  - nodeAnnotation: [undefined]
  - nodeName: Funnel
  - originalErrorMessage: Relative path is expected, path / is an absolute path.
  - stack: Error: Relative path is expected, path / is an absolute path.
    at Object.handleFs (/home/boris/project/node_modules/broccoli-funnel/node_modules/broccoli-output-wrapper/dist/index.js:27:15)
    at Funnel._copy (/home/boris/project/node_modules/broccoli-funnel/index.js:475:19)
    at Funnel.build (/home/boris/project/node_modules/broccoli-funnel/index.js:235:16)
    at /home/boris/project/node_modules/ember-cli/node_modules/broccoli/dist/wrappers/transform-node.js:68:45
    at new Promise (<anonymous>:null:null)
    at TransformNodeWrapper.build (/home/boris/project/node_modules/ember-cli/node_modules/broccoli/dist/wrappers/transform-node.js:53:16)
    at /home/boris/project/node_modules/ember-cli/node_modules/broccoli/dist/builder.js:99:36
```

**Comment by chancancode** [comment]
Probably related: in the ember-inspector build, we have:

```
// Stolen from relative-module-paths.js in ember-cli-babel
function getRelativeModulePath(modulePath) {
  console.log({ modulePath });
  return ensurePosix(path.relative(process.cwd(), modulePath));
}

// Stolen from relative-module-paths.js in ember-cli-babel
function resolveRelativeModulePath(name, child) {
  let relative = getRelativeModulePath(child);
  console.log({ name, child, relative });
  return moduleResolver(name, getRelativeModulePath(child));
}

// ...

let emberDebug = new Funnel('ember_debug', {
  destDir: 'ember-debug',
  include: ['**/*.js'],
  exclude: [
    'vendor/loader.js',
    'vendor/source-map.js',
    'vendor/startup-wrapper.js'
  ]
});

emberDebug = new Babel(emberDebug, {
  moduleIds: true,
  getModuleId: getRelativeModulePath,
  plugins: [
    ['module-resolver', { resolvePath: resolveRelativeModulePath }],
    ['transform-es2015-modules-amd', { noInterop: true }]
  ]
});
```

On 3.0.0, the logs are something like this:

```
{ modulePath: '/Code/ember-inspector/ember-debug/adapters/basic.js' }
{
  name: '../utils/on-ready',
  child: '/Code/ember-inspector/ember-debug/adapters/basic.js',
  relative: 'ember-debug/adapters/basic.js'
}
{ modulePath: '/Code/ember-inspector/ember-debug/adapters/basic.js' }
{ modulePath: '/Code/ember-inspector/ember-debug/adapters/basic' }
{ modulePath: '/Code/ember-inspector/ember-debug/adapters/chrome.js' }
{
  name: './web-extension',
  child: '/Code/ember-inspector/ember-debug/adapters/chrome.js',
  relative: 'ember-debug/adapters/chrome.js'
}
{ modulePath: '/Code/ember-inspector/ember-debug/adapters/chrome.js' }
{ modulePath: '/Code/ember-inspector/ember-debug/adapters/chrome' }
{
  modulePath: '/Code/ember-inspector/ember-debug/adapters/bookmarklet.js'
}
{
  name: './basic',
  child: '/Code/ember-inspector/ember-debug/adapters/bookmarklet.js',
  relative: 'ember-debug/adapters/bookmarklet.js'
}
...
```

On 3.0.1, it looks like this:

```
{ modulePath: '/Code/ember-inspector/adapters/basic.js' }
{
  name: '../utils/on-ready',
  child: '/Code/ember-inspector/adapters/basic.js',
  relative: 'adapters/basic.js'
}
{ modulePath: '/Code/ember-inspector/adapters/basic.js' }
{ modulePath: '/Code/ember-inspector/adapters/basic' }
{ modulePath: '/Code/ember-inspector/adapters/bookmarklet.js' }
{
  name: './basic',
  child: '/Code/ember-inspector/adapters/bookmarklet.js',
  relative: 'adapters/bookmarklet.js'
}
```

Note that the `modulePath` changed from something like `/Code/ember-inspector/ember-debug/adapters/basic.js` to `/Code/ember-inspector/adapters/basic.js`, which is wrong.

**Comment by makepanic** [comment]
We didn't get the error but our build behavior broke when upgrading from 3.0.0 to 3.0.1. too.

**Comment by boris-petrov** [comment]
This has been fixed in 3.0.3. Thanks!
